app/src/api/expectation.lisp

```diff
@@ -100,7 +100,8 @@ amplitudes in PREPARED-STATE."
     ;; generally smaller. Thus we compute tr(Q^T ρ^T) = tr((ρ Q)^T) = tr(ρ Q) = tr(Q ρ).
     (let ((op-matrix (magicl:transpose
                       (quil::parsed-program-to-logical-matrix op)))
-          (density-matrix (magicl:make-matrix :rows rows :cols cols 
-                                              :data prepared-state)))
-      (reduce #'+ (magicl:matrix-diagonal
+          (density-matrix (magicl:from-array prepared-state
+                                             (list rows cols)
+                                             :type '(complex double-float))))
```

Comment:
As long as the compiler doesn't complain, maybe the `:type` here should be `qvm:cflonum` instead? ditto elsewhere below.